### PR TITLE
Close popover on text field tap

### PR DESCRIPTION
**Pull request by Cyrine-benabid**
fixes [#1368](https://github.com/AppFlowy-IO/AppFlowy/issues/1368)

### CODE REVIEW

**Comment by annieappflowy** [comment]
@richardshiue , would you like to help review? This is a fix for [your issue](https://github.com/AppFlowy-IO/AppFlowy/issues/1368)

**Comment by richardshiue** [comment]
> @richardshiue , would you like to help review? This is a fix for [your issue](https://github.com/AppFlowy-IO/AppFlowy/issues/1368)

Sure, I'll review when I'm free

**Comment by richardshiue** [comment]
Hello @Cyrine-benabid, thanks for contributing and tackling this bug!

Your fix should close the popover correctly, but it's only going to happen by clicking the text field. This means that after I click on "Add option", the unwanted popover wouldn't disappear until I specifically click on the text field, even though the text field already has focus. A better solution would be to listen for focus changes in the popover's elements.

I will try and lay down some steps you can follow to finish this task.

File: frontend/app_flowy/lib/plugins/grid/presentation/widgets/common/text_field.dart [hunk 1]
```diff
@@ -18,6 +19,7 @@ class InputTextField extends StatefulWidget {
     this.onDone,
     required this.onCanceled,
     this.onChanged,
```
Comment (richardshiue) [inline frontend/app_flowy/lib/plugins/grid/presentation/widgets/common/text_field.dart:21]: I would add an optional FocusNode field here, and then in initState() down below, use the passed in FocusNode if it exists but create one if it doesn't. `_focusNode = widget.focusNode ?? FocusNode()`

File: frontend/app_flowy/lib/plugins/grid/presentation/widgets/header/type_option/select_option.dart [hunk 2]
```diff
@@ -273,6 +273,9 @@ class _CreateOptionTextField extends StatelessWidget {
                 .read<SelectOptionTypeOptionBloc>()
                 .add(SelectOptionTypeOptionEvent.createOption(optionName));
           },
+          onTap: () {
```
Comment (richardshiue) [inline frontend/app_flowy/lib/plugins/grid/presentation/widgets/header/type_option/select_option.dart:276]: Next, see if you can convert `_CreateOptionTextField` into a `StatefulWidget`, and allow passing in a `PopoverMutex`. Then, set up the `PopoverMutex` and `FocusNode` listeners. There's actually an extremely similar implementation [here](https://github.com/AppFlowy-IO/AppFlowy/blob/main/frontend/app_flowy/lib/plugins/grid/presentation/widgets/cell/date_cell/date_editor.dart#L298) that you can reference.

**Comment by Cyrine-benabid** [comment]
Hi @richardshiue ! Thank you for your review and your suggestion ! You can see in the video bellow my idea after the new commits. 
If this is not what you're looking for, don't hesitate to inform me. 
And thank you very much.
[fix_close_popover_on_click.webm](https://user-images.githubusercontent.com/98823356/198841829-92f74bd6-3fa5-4f52-befd-ef8d3b5c387d.webm)

**Comment by Cyrine-benabid** [comment]
Also @richardshiue, this was my original implementation (forgot to push one commit), but I also have a working implementation of your proposition in another commit. Let me know if you think we should use your proposition instead, and I'll push this commit.

**Comment by richardshiue** [comment]
Hello @Cyrine-benabid, thanks for the quick update. It solves the bug all right, no doubts there. But doing it the way I outlined above would also allow listening to the popover mutex to unrequest focus from the text field. I don't know that not supporting this will affect the UX much right now, if at all; but could be important if AppFlowy wants to improve keyboard accessibility in the future.

I think it's the perfect time to ask for @appflowy's views. Apart from the above concern, LGTM!

**Comment by Cyrine-benabid** [comment]
Hi @richardshiue, now I understand your point, thank you ! I pushed the implementation you suggested.

**Comment by richardshiue** [comment]
Almost there! `_AddOptionButton` doesn't seem need a popoverMutex either. Also, you can resolve the review comments above when you're done.

**Comment by richardshiue** [comment]
LGTM! Thanks for your work and patience with me during this process. Just need AppFlowy team's approval now.

**Comment by Cyrine-benabid** [comment]
Thanks to you @richardshiue for guiding me ! Looking forward to contribute more on this project :raised_hands: ! Hopefully this PR can be merged or hacktoberfest-accepted today :pray:

**Comment by annieappflowy** [comment]
Great work @Cyrine-benabid @richardshiue !

**Pull request merged**